src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj

```diff
@@ -88,6 +88,28 @@
       <Output TaskParameter="ItemsWithFullName" ItemName="_Dependency"/>
     </Microsoft.DotNet.Arcade.Sdk.GetAssemblyFullName>
 
+    <!-- Calculate metadata we need -->
+    <ItemGroup>
+      <_Dependency>
+        <_AssemblyName>$([MSBuild]::ValueOrDefault('%(_Dependency.FusionName)', '').Split(',')[0])</_AssemblyName>
+        <_AssemblyVersion>$([MSBuild]::ValueOrDefault('%(_Dependency.FusionName)', '').Split(',')[1].Split('=')[1])</_AssemblyVersion>
+        <_NuGetPackageFileName>%(_Dependency.NuGetPackageId).%(_Dependency.NuGetPackageVersion).nupkg</_NuGetPackageFileName>
+        <_NuGetPackageDir>$(NuGetPackageRoot)$([MSBuild]::ValueOrDefault('%(_Dependency.NuGetPackageId)', '').ToLower())\%(_Dependency.NuGetPackageVersion)\</_NuGetPackageDir>
+      </_Dependency>
+    </ItemGroup>
+
+    <!--
+      AssemblyVersionVariables include the dependency name and version that needs to be updated in the VS repo's
+      AssemblyVersions.tt file.  The roslyn insertion tool will read the output of this and update the file during the insertion.
+
+      These are separate from the _Dependency list as we do not need to always add the package dependency (e.g. private dependencies)
+    -->
+    <ItemGroup>
+      <_AssemblyVersionVariable Include="RoslynLanguageServerProtocol" Condition="'%(_Dependency.Identity)' == 'Microsoft.VisualStudio.LanguageServer.Protocol'">
+        <Version>%(_Dependency._AssemblyVersion)</Version>
+      </_AssemblyVersionVariable>
```

Comment:
I couldn't stick roslyn in here because the $(AssemblyVersion) variable isn't set, it needs to run in the target after GetAssemblyVersion

Reply:
You can add GetAssemblyVersion to the DependsOnTargets of _CalculateDependenciesToInsert

---
In reply to: [418351278](https://github.com/dotnet/roslyn/pull/43245#discussion_r418351278) [](ancestors = 418351278)

Reply:
🤦 fixed

Reply:
So now _GenerateDependentAssemblyVersions doesn't need GetAssemblyVersion :)